writer.go

```diff
@@ -335,6 +338,55 @@ func NewMediaPlaylist(winsize uint, capacity uint) (*MediaPlaylist, error) {
 	return p, nil
 }
 
+// InsertSegments allows the insertion of one or multiple MediaSegments into the MediaPlaylist.
+// seqID is the sequence ID at which the new segments should be inserted to
+func (p *MediaPlaylist) InsertSegments(segments []MediaSegment, seqID uint64) error {
+	if len(segments) == 0 {
+		return ErrPlaylistEmpty
+	}
+
+	// Create copies of the segments to be inserted
+	localSegments := make([]*MediaSegment, len(segments))
+	for i, segment := range segments {
+		copySegment := segment // Create a shallow copy of the segment
+		localSegments[i] = &copySegment
+	}
+
+	adjustment := uint(len(localSegments))
+	p.count += adjustment
+	p.tail += adjustment
+
+	// Determine the index where the new segments should be inserted
+	var insertIndex int
+	// Allow for insertion at the beginning of the playlist
+	if seqID == 0 {
+		insertIndex = 0
+	} else {
+		// Find the index where the new segments should be inserted
+		for i, segment := range p.Segments {
+			if segment.SeqId == seqID {
+				insertIndex = i + 1
+				break
+			}
+		}
+	}
+
+	// Insert the new segments into the playlist
+	p.Segments = append(p.Segments[:insertIndex], append(localSegments, p.Segments[insertIndex:]...)...)
```

Comment:
Not a big fan of double append, it can be inefficient, especially for large playlists maybe can use copy and shift the segments to create room?
```
	// Shift existing segments to make room for new ones
	copy(p.Segments[insertIndex+len(segments):], p.Segments[insertIndex:p.count])
```